File: ios/objc/StorylyDemo/StorylyDemo/CustomSizeViewController.m
//
//  UIViewController+CustomSizeViewController.m
//  StorylyDemo
//
//  Created by Haldun Melih Fadillioglu on 2.02.2021.
//  Copyright © 2021 App Samurai Inc. All rights reserved.
//

#import "CustomSizeViewController.h"

@interface CustomSizeViewController ()

@property (weak, nonatomic) IBOutlet StorylyView *storylyViewSmall;
@property (weak, nonatomic) IBOutlet StorylyView *storylyViewLarge;
@property (weak, nonatomic) IBOutlet StorylyView *storylyViewXLarge;
@property (weak, nonatomic) IBOutlet StorylyView *storylyViewPortrait;
@property (weak, nonatomic) IBOutlet StorylyView *storylyViewLandscape;

@end

@implementation CustomSizeViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    
    self.storylyViewSmall.storylyInit = [[StorylyInit alloc] initWithStorylyId: @"eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhY2NfaWQiOjc2MCwiYXBwX2lkIjo0MDUsImluc19pZCI6NDA0fQ.1AkqOy_lsiownTBNhVOUKc91uc9fDcAxfQZtpm3nj40"];
    self.storylyViewSmall.rootViewController = self;
//    self.storylyViewSmall.storyGroupSize = @"small";
    
    self.storylyViewLarge.storylyInit = [[StorylyInit alloc] initWithStorylyId: @"eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhY2NfaWQiOjc2MCwiYXBwX2lkIjo0MDUsImluc19pZCI6NDA0fQ.1AkqOy_lsiownTBNhVOUKc91uc9fDcAxfQZtpm3nj40"];
    self.storylyViewLarge.rootViewController = self;
//    self.storylyViewLarge.storyGroupSize = @"large";
    
    self.storylyViewXLarge.storylyInit = [[StorylyInit alloc] initWithStorylyId: @"eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhY2NfaWQiOjc2MCwiYXBwX2lkIjo0MDUsImluc19pZCI6NDA0fQ.1AkqOy_lsiownTBNhVOUKc91uc9fDcAxfQZtpm3nj40"];
    self.storylyViewXLarge.rootViewController = self;
//    self.storylyViewXLarge.storyGroupSize = @"xlarge";
    
    self.storylyViewPortrait.storylyInit = [[StorylyInit alloc] initWithStorylyId: @"eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhY2NfaWQiOjc2MCwiYXBwX2lkIjo0MDUsImluc19pZCI6NDA0fQ.1AkqOy_lsiownTBNhVOUKc91uc9fDcAxfQZtpm3nj40"];
    self.storylyViewPortrait.rootViewController = self;
//    self.storylyViewPortrait.storyGroupSize = @"custom";
    
//    self.storylyViewPortrait.storyGroupIconStyling = [[StoryGroupIconStyling alloc] initWithHeight:150 width:100 cornerRadius:15];
    
//    self.storylyViewPortrait.storyGroupIconHeight = 150;
//    self.storylyViewPortrait.storyGroupIconWidth = 100;
//    self.storylyViewPortrait.storyGroupIconCornerRadius = 15;
    
    self.storylyViewLandscape.storylyInit = [[StorylyInit alloc] initWithStorylyId: @"eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhY2NfaWQiOjc2MCwiYXBwX2lkIjo0MDUsImluc19pZCI6NDA0fQ.1AkqOy_lsiownTBNhVOUKc91uc9fDcAxfQZtpm3nj40"];
    self.storylyViewLandscape.rootViewController = self;
//    self.storylyViewLandscape.storyGroupSize = @"custom";
    
//    self.storylyViewLandscape.storyGroupIconStyling = [[StoryGroupIconStyling alloc] initWithHeight:100 width:150 cornerRadius:15];
    
//    self.storylyViewLandscape.storyGroupIconHeight = 100;
//    self.storylyViewLandscape.storyGroupIconWidth = 150;
//    self.storylyViewLandscape.storyGroupIconCornerRadius = 15;
}

@end

File: ios/swift/StorylyDemo/StorylyDemo/CustomSizeViewController.swift
//
//  CustomSizeViewController.swift
//  StorylyDemo
//
//  Created by Haldun Melih Fadillioglu on 28.01.2021.
//  Copyright © 2021 App Samurai Inc. All rights reserved.
//

import UIKit
import Storyly

class CustomSizeViewController: UIViewController {
    @IBOutlet weak var storylyViewSmall: StorylyView!
    @IBOutlet weak var storylyViewLarge: StorylyView!
    @IBOutlet weak var storylyViewXLarge: StorylyView!
    @IBOutlet weak var storylyViewPortrait: StorylyView!
    @IBOutlet weak var storylyViewLandscape: StorylyView!

    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        storylyViewSmall.storylyInit = StorylyInit(storylyId: @YOUR_APP_INSTANCE_TOKEN_FROM_DASHBOARD)
        storylyViewSmall.rootViewController = self
        //storylyViewSmall.storyGroupSize = "small"
        
        storylyViewLarge.storylyInit = StorylyInit(storylyId: @YOUR_APP_INSTANCE_TOKEN_FROM_DASHBOARD)
        storylyViewLarge.rootViewController = self
        //storylyViewLarge.storyGroupSize = "large"
        
        storylyViewXLarge.storylyInit = StorylyInit(storylyId: @YOUR_APP_INSTANCE_TOKEN_FROM_DASHBOARD)
        storylyViewXLarge.rootViewController = self
        //storylyViewXLarge.storyGroupSize = "xlarge"
        
        
        storylyViewPortrait.storylyInit = StorylyInit(storylyId: @YOUR_APP_INSTANCE_TOKEN_FROM_DASHBOARD)
        storylyViewPortrait.rootViewController = self
        //storylyViewPortrait.storyGroupSize = "custom"
        
        //storylyViewPortrait.storyGroupIconStyling = StoryGroupIconStyling(height: 150, width: 100, cornerRadius: 15)
        
         //storylyViewPortrait.storyGroupIconHeight = 150
         //storylyViewPortrait.storyGroupIconWidth = 100
         //storylyViewPortrait.storyGroupIconCornerRadius = 15
        
        storylyViewLandscape.storylyInit = StorylyInit(storylyId: @YOUR_APP_INSTANCE_TOKEN_FROM_DASHBOARD)
        storylyViewLandscape.rootViewController = self
        //storylyViewLandscape.storyGroupSize = "custom"
        
        //storylyViewLandscape.storyGroupIconStyling = StoryGroupIconStyling(height: 100, width: 150, cornerRadius: 15)
        
        //storylyViewLandscape.storyGroupIconHeight = 100
        //storylyViewLandscape.storyGroupIconWidth = 150
        //storylyViewLandscape.storyGroupIconCornerRadius = 15
    }

}
